### PR TITLE
Error message and stack trace logging not ECS compliant

### PR DESCRIPTION
**Pull request by moltenice**
With this change logs will look like:

```json
{
   "service":{
      "name":"petshop"
   },
   "@timestamp":"2020-05-11T15:01:01.000Z",
   "log":{
      "level":"error"
   },
   "message":"Artificially raised exception",
   "error":{
      "message":"divided by 0",
      "stack_trace":"..."
   }
}
```

### CODE REVIEW

File: lib/twiglet/logger.rb [hunk 1]
```diff
@@ -39,8 +39,8 @@ def warning(message)
     def error(message, error = nil)
       if error
         message = message.merge({
-                                  error_name: error.message,
-                                  backtrace: error.backtrace
+                                  'error.message': error.message,
```
Comment (sleepyfox) [inline lib/twiglet/logger.rb]:
```suggestion
                                  error: { message: error.message,
                                           stack_trace: error.backtrace.join("\n") }
```

**Review by sleepyfox — CHANGES_REQUESTED** [review]
We shouldn't be encouraging the use of 'dotted keys'.

File: lib/twiglet/logger.rb (hunk 1 above)
Comment (sleepyfox) [reply lib/twiglet/logger.rb]: Worthwhile also adding an example of how to log an error in the readme and example file.

**Comment by moltenice** [comment]
> We shouldn't be encouraging the use of 'dotted keys'.

Our README.md definitely encourages its use :)

https://github.com/simplybusiness/twiglet-ruby#use-of-dotted-keys

**Comment by sleepyfox** [comment]
> > We shouldn't be encouraging the use of 'dotted keys'.
> 
> Our README.md definitely encourages its use :)

It's there as a backup, because Quynh felt strongly that we should help people acclimatise to the use of structured logging. Think of it as a training wheel. My view was that it was a needless complication and they should just rip off the band-aid, but I guess I'm just...

**Review by sleepyfox — APPROVED** [review]
:shipit:

File: lib/twiglet/logger.rb (hunk 1 above)
Comment (moltenice) [reply lib/twiglet/logger.rb]: Done both. Example file already had this but added it to README.md

**Pull request merged**